### PR TITLE
[RPC] Use "shield" nomenclature instead of "shielded"

### PR DESCRIPTION
**Pull request by Fuzzbawls**
Alternative to #2077 that limits the scope to the RPC interface (and unit/functional tests that exercise the RPC interface)

Drop the "ed" suffix in the RPC scope, changes commands like
`getnewshieldedaddress` to `getnewshieldaddress`.

Also in blockchain informational calls; `shielded_pool_value` and
`shieldedsupply` become `shield_pool_value` and `shieldsupply`
respectively.

### CODE REVIEW

File: src/rpc/misc.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ UniValue getinfo(const JSONRPCRequest& request)
             "  \"transparentsupply\" : n       (numeric) The sum of the value of all unspent outputs when the chainstate was\n"
             "                                            last flushed to disk (use getsupplyinfo to know the update-height, or\n"
             "                                            to trigger the money supply update/recalculation)"
-            "  \"shieldedsupply\": n           (numeric) Chain tip shielded pool value\n"
+            "  \"shieldsupply\": n             (numeric) Chain tip shield pool value\n"
```
Comment (random-zebra) [inline src/rpc/misc.cpp:74]: we need to also update line 136.
Both the key pushed in the returned object here, and the one of `supply_info`, returned from `getsupplyinfo`, from where we pick the value.

**Review by random-zebra — CHANGES_REQUESTED** [review]
there is a bug in `getinfo`, returning
```
"shieldedsupply": null,
```

instead of
```
"shieldsupply": 3452378.06910630,
```
as per `getsupplyinfo` output.

File: src/rpc/misc.cpp (hunk 1 above)
Comment (furszy) [reply src/rpc/misc.cpp:74]: Hmm, why `sapling_supply.py` is passing?

Comment (random-zebra) [reply src/rpc/misc.cpp:74]: because it calls directly `getsupplyinfo`. Not `getinfo`.

Comment (furszy) [reply src/rpc/misc.cpp:74]: kk, all good then.

Comment (Fuzzbawls) [reply src/rpc/misc.cpp:74]: fixed

**Review by random-zebra — APPROVED** [review]
utACK 51db2881d1c8ef424e8010b12975865fa5651a8f

**Review by furszy — APPROVED** [review]
utACK 51db288

**Pull request merged**